examples/fabric-bridge-app/linux/DeviceManager.cpp

```diff
@@ -81,8 +154,9 @@ int DeviceManager::AddDeviceEndpoint(Device * dev, EmberAfEndpointType * ep,
                     emberAfSetDynamicEndpoint(index, mCurrentEndpointId, ep, dataVersionStorage, deviceTypeList, parentEndpointId);
                 if (err == CHIP_NO_ERROR)
                 {
-                    ChipLogProgress(NotSpecified, "Added device %s to dynamic endpoint %d (index=%d)", dev->GetName(),
-                                    mCurrentEndpointId, index);
+                    ChipLogProgress(NotSpecified,
+                                    "Added device with nodeId=0x" ChipLogFormatX64 " to dynamic endpoint %d (index=%d)",
+                                    ChipLogValueX64(dev->GetNodeId()), mCurrentEndpointId, index);
```

Comment:
In all the error cases below where you return -1, you are not setting mDevices[index] back to null. Is this a problem?

Reply:
Good catch, if we do not set mDevices[index] back to nullptr when returning -1 due to an error, it can leave the mDevices array in an inconsistent state, where mDevices[index] points to a device that wasn't properly added.

This issue also exist in Bridge Example App Where I copy from, https://github.com/project-chip/connectedhomeip/issues/33946